streamer/src/nonblocking/quic.rs

```diff
@@ -473,10 +521,48 @@ impl ConnectionTable {
         num_pruned
     }
 
+    fn connection_stake(&self, index: usize, default: u64) -> u64 {
+        self.table
+            .get_index(index)
+            .map_or(default, |(_, connection_vec)| {
+                connection_vec
+                    .first()
+                    .map_or(default, |connection| connection.stake)
+            })
+    }
+
+    // Randomly select two connections, and evict the one with lower stake. If the stakes of both
+    // the connections are higher than the threshold_stake, reject the pruning attempt, and return 0.
+    fn prune_random(&mut self, threshold_stake: u64) -> usize {
+        let mut num_pruned = 0;
+        let mut rng = thread_rng();
+        let candidate1 = rng.gen_range(0, self.table.len());
+        let candidate2 = rng.gen_range(0, self.table.len());
```

Comment:
These two candidates could be the same, I think that's actually fine but should add a comment about it